.github/workflows/update-dependencies.yml

```
@@ -3,92 +3,187 @@ name: Update dependencies
-
-          - key: kayobe
-            path: src/kayobe-config/requirements.txt
-            repository: stackhpc/kayobe
-            search_regex: 'kayobe@stackhpc\/.*$'
-            prefix: 'kayobe@'
+    runs-on: ubuntu-24.04
     permissions:
       contents: write
       pull-requests: write
-    name: ${{ matrix.key }}
+    name: Propose dependency updates
+    outputs:
+      kolla-tag: ${{ steps.latest_kolla_tag.outputs.latest_tag || steps.current_kolla_version.outputs.version }}
+      kolla-ansible-tag: ${{ steps.latest_kolla_ansible_tag.outputs.latest_tag || steps.current_kolla_ansible_version.outputs.version }}
+      kayobe-tag: ${{ steps.latest_kayobe_tag.outputs.latest_tag || steps.current_kayobe_version.outputs.version }}
     steps:
-      - name: Checkout
+      - name: Checkout Kayobe-config
         uses: actions/checkout@v4
         with:
-          ref: ${{ inputs.openstack_version }}
-          path: ${{ github.workspace }}/src/kayobe-config
+          ref: ${{ inputs.branch }}
+          path: src/kayobe-config
+
+      - name: Set sanitised branch name
+        id: branch_name
+        run: |
+          sanitised_name=$(echo "update-dependencies-${{ inputs.branch }}" | tr '/' '-')
+          echo "name=${sanitised_name}" >> $GITHUB_OUTPUT
+
+      - name: Set up branch and Git config
+        run: |
+          git checkout -b ${{ steps.branch_name.outputs.name }}
+          git config user.name "stackhpc-ci"
+          git config user.email "22933334+stackhpc-ci@users.noreply.github.com"
+        working-directory: src/kayobe-config
+
+      - name: Initialise PR Body
+        run: |
+          echo "This PR was created automatically to update dependencies for the ${{ inputs.branch }} release." > pr_body.md
+          echo "" >> pr_body.md
+          echo "### Changes" >> pr_body.md
+
+      - name: Checkout Kolla repository
+        uses: actions/checkout@v4
+        with:
+          repository: stackhpc/kolla
+          ref: ${{ inputs.branch }}
+          fetch-tags: true
+          path: src/kolla
+
+      - name: Get latest Kolla tag
+        id: latest_kolla_tag
+        run: echo "latest_tag=$(git describe --tags --abbrev=0 --match stackhpc/\*)" >> $GITHUB_OUTPUT
+        working-directory: ${{ github.workspace }}/src/kolla
+
+      - name: Get current Kolla version
+        id: current_kolla_version
+        run: |
+          VERSION=$(awk -F': ' '/stackhpc_kolla_source_version:/ {print $2}' src/kayobe-config/etc/kayobe/stackhpc.yml | xargs)
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
 
-      - name: Checkout the dependency repo
+      - name: Update and commit Kolla version if needed
+        if: steps.latest_kolla_tag.outputs.latest_tag != steps.current_kolla_version.outputs.version
+        run: |
+          sed -i "s/stackhpc_kolla_source_version\:.*$/stackhpc_kolla_source_version\: $(echo $TAG | sed 's/\//\\\//g')/g" etc/kayobe/stackhpc.yml
```

```suggestion
          sed -i "s/stackhpc_kolla_source_version\:.*$/stackhpc_kolla_source_version\: $(echo $TAG | sed 's|/|\\/|g')/g" etc/kayobe/stackhpc.yml
```
maybe changing the sed delimiter makes it a bit more readable?
It has been done for kayobe

---

Nice idea, that makes a lot of sense